types/filters/parse.go

```diff
@@ -80,7 +81,7 @@ func ToParamWithVersion(version string, a Args) (string, error) {
 	// for daemons older than v1.10, filter must be of the form map[string][]string
 	buf := []byte{}
 	err := errors.New("")
-	if version != "" && compareTo(version, "1.22") == -1 {
+	if version != "" && versions.LessThan(version, "1.22") {
```

Comment:
can we cast version to pkg/version and do the check with pkg/version code instead?